### DISCUSSION TITLE
How to edit/change the measurement text of an angular dimension?

### DISCUSSION THREAD

**Discussion by Mehdi1921** [mozman/ezdxf#276] [Q&A]
@mozman 
Dear Mr. Moitzi,

Thank you very much for preparation of the fantastic EZDXF module. 
Could you please inform me how I can add a user defined text (e.g., text=str(self.angle)) to the AngularDimension function? 

I have tried dimstyles.new but I could not add the defined text. 

![image](https://user-images.githubusercontent.com/61479664/101890963-b78a4e00-3ba1-11eb-98eb-aeb247777ee1.png)

Thank you!

**Comment by Mehdi1921** [comment]
In this case, I would like to define a text (35 °) and replace it to -567°.

**Comment by mozman** [comment]
As the [documentation](https://ezdxf.mozman.at/docs/dxfentities/dimension.html) says, adding an angular dimension is not implemented yet. 

To edit an existing angular dimension you have to modify the content of the associated anonymous block, which represents the dimension content as DXF primitives, the name of this block is stored in `Dimension.dxf.geometry` attribute (search for a `TEXT` or a `MTEXT` entity. The displayed dimension text is stored in `Dimension.dxf.text`, but editing this attribute has only an effect when the dimension is rendered again, which is not implemented in ezdxf yet.

And to repeat myself over and over again, ezdxf is not a CAD application, this is not the intended usage of ezdxf.
